src/mini_s3.erl

```diff
@@ -115,327 +131,153 @@ manual_start() ->
+        % Host == domain
+        [Domain1] ->
+            Scheme = "https://",
+            Port1  = undefined
+    end,
+    Port =
+        case Port1 of
+            undefined ->
+                case Scheme of
+                    "https://" -> 443;
+                    "http://"  -> 80
+                end;
+            _ ->
+                list_to_integer(Port1)
+        end,
+    Domain = case Ipv of 4 -> Domain1; _ -> "[" ++ Domain0 ++ "]" end,
+    %% bookshelf wants bucketname after host e.g. https://api.chef-server.dev:443/bookshelf...
+    %% s3 wants bucketname before host (or it takes it either way) e.g. https://bookshelf.api.chef-server.dev:443...
+    %% amazon: "Buckets created after September 30, 2020, will support only virtual hosted-style requests.
+    %% Path-style requests will continue to be supported for buckets created on or before this date."
+    %% for further discussion, see:
+    %%  https://github.com/chef/chef-server/issues/2088
+    %%  https://github.com/chef/chef-server/issues/1911
+
+    (erlcloud_s3:new(AccessKeyID, SecretAccessKey, Domain, Port))#aws_config{s3_scheme=Scheme, s3_bucket_after_host=true, s3_bucket_access_method=path}.
+
+% old mini_s3 wanted accesskey, secretaccesskey, host, bucketaccesstype, e.g.:
+%   -spec new(string(), string(), string(), bucket_access_type()) -> aws_config().
+% erlcloud wants accesskey, secretaccesskey, host, port.
+-spec new(string() | binary(), string() | binary(), string(), bucket_access_type()) -> aws_config().
 new(AccessKeyID, SecretAccessKey, Host, BucketAccessType) ->
-    #config{
-     access_key_id=AccessKeyID,
-     secret_access_key=SecretAccessKey,
-     s3_url=Host,
-     bucket_access_type=BucketAccessType}.
-
--spec new(string(), string(), string(), bucket_access_type(), proplists:proplist()) -> config().
-new(AccessKeyID, SecretAccessKey, Host, BucketAccessType, SslOpts) ->
-    #config{
-     access_key_id=AccessKeyID,
-     secret_access_key=SecretAccessKey,
-     s3_url=Host,
-     bucket_access_type=BucketAccessType,
-     ssl_options=SslOpts}.
-
-
+    {BucketAccessMethod, BucketAfterHost} = case BucketAccessType of path -> {path, true}; _ -> {vhost, false} end,
+    Config = new(AccessKeyID, SecretAccessKey, Host),
+    Config#aws_config{
+        s3_bucket_access_method=BucketAccessMethod,
+        s3_bucket_after_host=BucketAfterHost
+    }.
+
+% erlcloud has no new/5, and arguments differ.
+% erlcloud's new/4 expects accesskeyid, secretaccesskey, host, port:
+%   new(AccessKeyID::string(), SecretAccessKey::string(), Host::string(), Port::non_neg_integer()) -> aws_config()
+% for now, attempting conversion to new/4 (dropping SslOpts).
+%
+% NOTE: this is called in oc_erchef in: src/oc_erchef/apps/chef_objects/src/chef_s3.erl, line 168
+-spec new(string() | binary(), string() | binary(), string(), bucket_access_type(), proplists:proplist()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, BucketAccessType, _SslOpts) ->
```

Comment:
Dropping SSL Options is going to be a problem for A2-based Chef Server's as we need to be able to pass the SSL Options to talk to bookshelf with the right TLS configuration.

Reply:
i'll have to investigate this.

Reply:
@PrajaktaPurohit and I did a 'mini triage' on this.  She seems to think we can kick the can down the road for a bit, and revisit later (possibly even after a merge).  Do you concur?

Reply:
I think since it is 2 months past that discussion - now is a good time to revisit this issue :) We should probably create a ticket if we haven't already for this.

Reply:
Ticket created.